### PR TITLE
feature/MRTI-3464 # hashkeys and uids selection DON'T MERGE ME

### PR DESCRIPTION
**Pull request by juan-kabbali**

### CODE REVIEW

File: datalake_scripts/scripts/iocs_select_uids.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    :return: (bool, str)
+
+    'hashkeys-2021-01-31-2021-04-13-malware-ip-10-80.txt'                             relative
+    '/some/path/to_file/hashkeys-2021-01-31-2021-04-13-malware-ip-10-80.txt'          absolut
+    """
+    is_valid, validation_msg = True, 'ok'
+    input_filename_regex_validation = r'hashkeys-' \
+                                      r'(\d{4}-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])-?){2}-' \
+                                      r'(\w+-?){2}-' \
+                                      r'(\d{1,2}-?){2}\.txt$'
```
Comment (juan-kabbali) [inline datalake_scripts/scripts/iocs_select_uids.py:161]: this won't match the number `100` when the score is the maximum

File: datalake_scripts/scripts/iocs_select_uids.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+            'first_seen'
+        ],
+        'hashkeys': args.hashkeys
+    }
+
+
+def _make_output_file_name(args, threat_type, atom_type, min_score, max_score):
+    date_range = f'{args.created_since.strftime(DATE_FORMAT)}-{args.created_until.strftime(DATE_FORMAT)}'
+    score_range = f'{min_score}-{max_score}'
+    return f'iocs-{date_range}-{threat_type}-{atom_type}-{score_range}.txt'
```
Comment (juan-kabbali) [inline datalake_scripts/scripts/iocs_select_uids.py:187]: use underscore as field separator

File: datalake_scripts/scripts/iocs_select_hashkeys.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+            }
+        ]
+    }
+
+
+def _make_output_file_name(args):
+    date_range = f'{args.from_date}-{args.to_date}'
+    score_range = f'{args.min_score}-{args.max_score}'
+
+    return f'hashkeys-{date_range}-{args.threat_type}-{args.atom_type}-{score_range}.txt'
```
Comment (juan-kabbali) [inline datalake_scripts/scripts/iocs_select_hashkeys.py:174]: use underscore as field separator

File: datalake_scripts/engines/get_engine.py [hunk 4]
```diff
@@ -106,3 +107,44 @@ def get_threats(self, query_hash: str, limit=10, response_format="application/js
...
+
+    def _build_url(self, endpoint_config: dict, environment: str):
+        return self._build_url_for_endpoint('events')
+
+    def get_events(
+            self,
+            hashkeys: List,
+            limit=10,
+            offset=0,
+            ordering=[],
```
Comment (Hugo-C) [inline datalake_scripts/engines/get_engine.py:125]: Better to use `None` even if we don't write in it: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

**Review by Hugo-C — APPROVED** [review]

**Comment by juan-kabbali** [comment]
All iocs has been collected and the command works properly on their own branch `feature/MRTI-3464`

**Pull request closed without merging**